File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/message/MessageEntity.java
package com.github.dedis.popstellar.repository.database.message;

import androidx.annotation.NonNull;
import androidx.room.*;

import com.github.dedis.popstellar.model.Immutable;
import com.github.dedis.popstellar.model.network.method.message.MessageGeneral;
import com.github.dedis.popstellar.model.objects.security.MessageID;

import java.util.Objects;

import javax.annotation.Nullable;

@Entity(tableName = "messages")
@Immutable
public class MessageEntity {

  @ColumnInfo(name = "message_id")
  @PrimaryKey
  @NonNull
  private final MessageID messageId;

  @ColumnInfo(name = "message")
  @Nullable
  private final MessageGeneral content;

  public MessageEntity(@NonNull MessageID messageId, @Nullable MessageGeneral content) {
    this.messageId = messageId;
    this.content = content;
  }

  @NonNull
  public MessageID getMessageId() {
    return messageId;
  }

  @Nullable
  public MessageGeneral getContent() {
    return content;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    MessageEntity that = (MessageEntity) o;
    return messageId.equals(that.messageId) && Objects.equals(content, that.content);
  }

  @Override
  public int hashCode() {
    return Objects.hash(messageId, content);
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/message/MessageEntity.kt
package com.github.dedis.popstellar.repository.database.message

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.github.dedis.popstellar.model.Immutable
import com.github.dedis.popstellar.model.network.method.message.MessageGeneral
import com.github.dedis.popstellar.model.objects.security.MessageID
import java.util.Objects

@Entity(tableName = "messages")
@Immutable
class MessageEntity(
    @field:PrimaryKey @field:ColumnInfo(name = "message_id") val messageId: MessageID,
    @field:ColumnInfo(name = "message") val content: MessageGeneral?
) {

  override fun equals(other: Any?): Boolean {
    if (this === other) {
      return true
    }
    if (other == null || javaClass != other.javaClass) {
      return false
    }
    val that = other as MessageEntity
    return messageId == that.messageId && content == that.content
  }

  override fun hashCode(): Int {
    return Objects.hash(messageId, content)
  }
}
